stacks/databaseMigrations.ts

```diff
@@ -1,9 +1,17 @@
 import { StackContext, use } from '@serverless-stack/resources';
 import { Network } from 'stacks/network';
+import { Database } from './database';
 import { DbMigrationScript } from './resources/migrationScript';
+import { WakeDB } from './resources/wakeUpDB';
 
 export function DatabaseMigrations({ stack }: StackContext) {
   const net = use(Network);
+  const { cluster } = use(Database);
 
-  new DbMigrationScript(stack, 'MigrationScript', { vpc: net.vpc });
+  // make sure DB is running before applying migration
+  const wakeUp = new WakeDB(stack, 'WakeDB', { cluster });
```

Comment:
I find it counter intuitive that the wakeUp method doesn't wait for it to wake up and that instead it's delegated to the migration method, couldn't we run a simple inexpensive check command in the wake up method and sleep there?

Reply:
Yeah that would be better. Requires writing an actual function though. Trying to be lazy here but the retry logic in the migration script was not really ideal so idk.

Reply:
I mean it doesn't really make a difference it's just how it works internally so not a big deal anyway

Reply:
I'd like to try improving it one day